tests/plugins/language/language.js

```diff
@@ -144,6 +144,123 @@
 			);
 			this.editor.execCommand( 'removeFormat' );
 			assert.beautified.html( '<p><span dir="ltr" lang="fr">Lorem ipsum dolor somit</span></p>', this.editor.getData(), 'Span element with atrributes should not be removed.' );
+		},
+
+		'test coerces a correct value when the \'dir\' attribute is empty in RTL language': function() {
+			var editor = this.editor,
+				language = 'ar';
+
+			editor.setMode( 'source', function() {
+				resume( function() {
+					editor.editable().setValue( '<p>foo <span lang="' + language + '" dir="">bar</span></p>' );
+					editor.setMode( 'wysiwyg', function() {
+						resume( function() {
+							var expected = '';
+
+							if ( CKEDITOR.env.gecko || ( CKEDITOR.env.ie && CKEDITOR.env.version < 10 ) ) {
+								expected = '<p>foo <span dir="rtl" lang="' + language + '">bar</span></p>';
+							} else {
+								expected = '<p>foo <span lang="' + language + '" dir="rtl">bar</span></p>';
+							}
+
+							assert.areSame(
+								editor.editable().getData(), expected,
+								'The value of dir attribute for ' + language + ' language is incorrect.'
+							);
```

Comment:
Can't it be simplified by using `assert.beautified.html()`? It has an option to sort attributes, so we won't need two distinctive versions with all the same but attributes order

Reply:
Yes, it can be used here, good point. 👍  Tests are already updated.